scripts/test/lint.py

```diff
@@ -199,22 +207,32 @@ class VersionNotUpdated(Lint):
     run_cmd(f"{GIT_EXE} version")
     run_cmd(f"{GIT_EXE} --no-pager branch -r")
     changed_files = run_cmd(f"{GIT_EXE} --no-pager diff --name-only {GITHUB_BASE_REF}")
-    changed_files = set(changed_files.splitlines())
-    nuspecs = set([file for file in changed_files if file.endswith(".nuspec")])
+    changed_files = set(map(pathlib.Path, changed_files.splitlines()))
+    nuspecs = set([file for file in changed_files if file.suffix == ".nuspec"])
     others = changed_files - nuspecs
 
     logger.debug("changed: %s", changed_files)
     logger.debug("nuspecs: %s", nuspecs)
     logger.debug("others: %s", others)
 
     def check(self, path):
+        package_path = None
         for part in path.parts:
             if part.endswith(".vm"):
-                # only check exact package path, i.e., `/<package>/`
-                package_path = f"{os.sep}{part}{os.sep}"
+                # find package path, only want to check exact path, i.e., `/<package>/`
+                # note: git appears to return slash (/) separated paths on Windows and Linux
+                # working around this here via pathlib
+                package_path = f"{part}"
```

Comment:
main bug was here before due to use of `os.sep` whereas git always returned paths with `/`, using pathlib now to hopefully work reliably across systems